src/main/java/donkeychat/Parser.java

```diff
@@ -0,0 +1,130 @@
+                    } catch (NumberFormatException ex) {
+                        ex.printStackTrace();
+                        break;
+                    }
+                    if (taskIndex < 0 || taskIndex > taskList.getSize()) {
+                        System.out.println("invalid task index!");
+                        break;
+                    }
+                    task = taskList.getAtIndex(taskIndex);
+                    task.setDone(false);
+                    ui.displayMarkTask(false, task);
+                    break;
+                case "todo":
+                    if (splitInput.length == 1) {
+                        throw new DukeException("'todo' requires additional arguments!");
+                    }
+                    taskList.addTask(new ToDo(splitInput[1]));
+                    ui.displayAddTask(taskList);
+                    break;
+                case "deadline":
+                    if (splitInput.length == 1) {
+                        throw new DukeException("'deadline' requires additional arguments!");
+                    }
+                    Integer indexBy = splitInput[1].indexOf("/by ");
+                    taskList.addTask(
+                        new Deadline(splitInput[1].substring(0, indexBy - 1),
+                            splitInput[1].substring(indexBy + 4)));
+                    ui.displayAddTask(taskList);
+                    break;
+                case "event":
+                    if (splitInput.length == 1) {
+                        throw new DukeException("'event' requires additional arguments!");
+                    }
+                    Integer indexFrom = splitInput[1].indexOf("/from ");
+                    Integer indexTo = splitInput[1].indexOf("/to ");
+                    taskList.addTask(new Event(
+                        splitInput[1].substring(0, indexFrom - 1),
+                        splitInput[1].substring(indexFrom + 6, indexTo - 1),
+                        splitInput[1].substring(indexTo + 4)));
+                    ui.displayAddTask(taskList);
+                    break;
+                case "delete":
+                    if (splitInput.length == 1) {
+                        throw new DukeException("'delete' requires additional arguments!");
+                    }
+                    try {
+                        taskIndex = Integer.valueOf(splitInput[1]) - 1;
+                    } catch (NumberFormatException ex) {
+                        ex.printStackTrace();
+                        break;
+                    }
+
+                    if (taskIndex < 0 || taskIndex > taskList.getSize()) {
+                        System.out.println("invalid task index!");
+                        break;
+                    }
+                    ui.displayDeleteTask(taskList.getAtIndex(taskIndex));
+                    taskList.removeAtIndex(taskIndex.intValue());
+                    break;
+                case "find":
```

Comment:
I think your code is quite easy to follow! Perhaps u might want to align the switch and case to follow the coding standard! You can also consider declaring the commands (find, mark, todo ...) as constants and use it at the case statement as well!